packages/component-base/src/gestures.js

```diff
@@ -0,0 +1,1096 @@
+  let dx = lastmove.x - info.x;
+  let dy = lastmove.y - info.y;
+  let ddx,
+    ddy = 0;
+  if (secondlast) {
+    ddx = lastmove.x - secondlast.x;
+    ddy = lastmove.y - secondlast.y;
+  }
+  _fire(target, 'track', {
+    state: info.state,
+    x: touch.clientX,
+    y: touch.clientY,
+    dx: dx,
+    dy: dy,
+    ddx: ddx,
+    ddy: ddy,
+    sourceEvent: touch,
+    hover: function () {
+      return deepTargetFind(touch.clientX, touch.clientY);
+    }
+  });
+}
+
+register({
+  name: 'tap',
+  deps: ['mousedown', 'click', 'touchstart', 'touchend'],
+  flow: {
+    start: ['mousedown', 'touchstart'],
+    end: ['click', 'touchend']
+  },
+  emits: ['tap'],
+  info: {
+    x: NaN,
+    y: NaN,
+    prevent: false
+  },
+  /**
+   * @this {GestureRecognizer}
+   * @return {void}
+   */
+  reset: function () {
+    this.info.x = NaN;
+    this.info.y = NaN;
+    this.info.prevent = false;
+  },
+  /**
+   * @this {GestureRecognizer}
+   * @param {MouseEvent} e
+   * @return {void}
+   */
+  mousedown: function (e) {
+    if (hasLeftMouseButton(e)) {
+      this.info.x = e.clientX;
+      this.info.y = e.clientY;
+    }
+  },
+  /**
+   * @this {GestureRecognizer}
+   * @param {MouseEvent} e
+   * @return {void}
+   */
+  click: function (e) {
+    if (hasLeftMouseButton(e)) {
+      trackForward(this.info, e);
+    }
+  },
+  /**
+   * @this {GestureRecognizer}
+   * @param {TouchEvent} e
+   * @return {void}
+   */
+  touchstart: function (e) {
+    const touch = e.changedTouches[0];
+    this.info.x = touch.clientX;
+    this.info.y = touch.clientY;
+  },
+  /**
+   * @this {GestureRecognizer}
+   * @param {TouchEvent} e
+   * @return {void}
+   */
+  touchend: function (e) {
+    trackForward(this.info, e.changedTouches[0], e);
+  }
+});
+
+/**
+ * @param {!GestureInfo} info
+ * @param {Event | Touch} e
+ * @param {Event=} preventer
+ * @return {void}
+ */
+function trackForward(info, e, preventer) {
+  let dx = Math.abs(e.clientX - info.x);
+  let dy = Math.abs(e.clientY - info.y);
+  // find original target from `preventer` for TouchEvents, or `e` for MouseEvents
+  let t = _findOriginalTarget(preventer || e);
+  if (!t || (canBeDisabled[/** @type {!HTMLElement} */ (t).localName] && t.hasAttribute('disabled'))) {
+    return;
+  }
+  // dx,dy can be NaN if `click` has been simulated and there was no `down` for `start`
+  if (isNaN(dx) || isNaN(dy) || (dx <= TAP_DISTANCE && dy <= TAP_DISTANCE) || isSyntheticClick(e)) {
+    // prevent taps from being generated if an event has canceled them
+    if (!info.prevent) {
+      _fire(t, 'tap', {
+        x: e.clientX,
+        y: e.clientY,
+        sourceEvent: e,
+        preventer: preventer
+      });
+    }
+  }
+}
+
+/* eslint-enable valid-jsdoc */
+
+/** @deprecated */
+export const findOriginalTarget = _findOriginalTarget;
+
+/** @deprecated */
+export const add = addListener;
+
+/** @deprecated */
+export const remove = removeListener;
```

Comment:
Let's consider removing these deprecated exports as we have no use for them in our codebase.

Reply:
Done